content/post/2021-06-28-draw-me-a-project/index.md

```diff
@@ -0,0 +1,267 @@
+
+Now why should *I* discuss the making of R projects? While I am not often in charge of analyses these days, I [follow R news](/2019/01/25/uptodate/) quite well so thought I might be able to deliver some useful tips to the audience.
+
+## Some "basics"
+
+To start with I'd like to mention some rules or principles that are quite crucial. My main source of information for them is [Jenny Bryan](https://jennybryan.org/).
+
+{{< tweet 1222960316545294337 >}}
+
+Like many I identified with Sharla Gelfand's quote[^2] as reported by Kara Woo on Twitter, "Everything I know is from Jenny Bryan".
+
+{{< figure src="https://raw.githubusercontent.com/MonkmanMH/EIKIFJB/main/EIKIFJB_sigmar_hex.png" alt="hex logo de 'Everything I know is from Jenny Bryan' avec un ordinateur portable en feu" width=350 link="https://github.com/MonkmanMH/EIKIFJB" >}}
+
+There's even a [logo](https://github.com/MonkmanMH/EIKIFJB) created by [Martin Monkman](https://github.com/MonkmanMH) for all of us agreeing with that statement! It features the letters EIKIFJB for "Everything I know is from Jenny Bryan" as well as a laptop on fire.
+
+Why a laptop on fire?!
+
+{{< tweet 940021008764846080 >}}
+
+That's a reference to a talk Jenny Bryan gave years ago, to which she said she'd put your computer on fire if you used `setwd("C:\Users\jenny\path\that\only\I\have")` or `rm(list = ls())` at the beginning of your scripts. This goes against the notion of a ["project-oriented workflow"](https://www.tidyverse.org/blog/2017/12/workflow-vs-script/). I do not need to repeat the wisdom of her blog post in mine, so will only use cliff notes:
+
+-   When reading in data etc. use paths relative to the root of your project, possibly using the [here package by Kirill Müller](https://github.com/jennybc/here_here).
+-   Re-start R often, do not e.g. load packages in your [.Rprofile](https://rstats.wtf/r-startup.html#rprofile), [`usethis::use_blank_slate()`](https://usethis.r-lib.org/reference/use_blank_slate.html).
+
+Really, go and read her post if you haven't yet!
+
+In my talk I'll show pictures of the Kei-tora Gardening Contest where participants create beautiful gardens in little trucks, which I find is a good image of a project that should be independent (you can move your truck, you should be able to move your project too).
+
+{{< tweet 1403009688392781824 >}}
+
+As the first part of my talk really is a collection of awesome content by Jenny Bryan, I will also show her rules for [naming files](http://www2.stat.duke.edu/~rcs46/lectures_2015/01-markdown-git/slides/naming-slides/naming-slides.pdf). What Shakespeare said about [roses](https://en.wikipedia.org/wiki/A_rose_by_any_other_name_would_smell_as_sweet) isn't true when programming. 😅
+
+Then I'll "introduce" version control. I'll briefly mention some basic git commands, and most importantly, how to run them! Here are my preferences:
```

Comment:
It is interesting that you introduce Version Control before talking about any structure organisation of a project. Or at least that Project is tied to IDE and this is why IDE are so widely used by developer - because the enforce and help with Project management in the first place. Git comes second to me, just because you need to create a project structure before doing git. 

1. Create project : `usethis::create_project()` - mandatory always ! 
2. Setup version control: `usethis::use_git()` - really really advice ! Local safeguard for time capsule and more
3. Save in the cloud, share and collaborate: `usethis::use_github()` - New user must see this as a safety net in case of computer crash 😄 - Gitlab and other are also valid option
4. Deps management - time capsule with renv - requires a project too !

Anyway, I would have highlighted more about how project can be created and tied to IDE 

* https://rstats.wtf/project-oriented-workflow.html#ide-support-for-projects
* https://rstats.wtf/project-oriented-workflow.html#rstudio-projects